Python/symtable.c

```diff
@@ -467,8 +473,7 @@ analyze_name(PySTEntryObject *ste, PyObject *scopes, PyObject *name, long flags,
     if (flags & DEF_GLOBAL) {
         if (flags & DEF_PARAM) {
```

Comment:
Is this check still needed?

Reply:
Hm, indeed this looks redundant. Initially I thought that it is used to prohibit:
```python
def f(x):
    def g():
        global x
```
But this is not an error, and after thinking more it looks like it should not be an error, since ``global`` and ``nonlocal`` have unwanted interference with being parameter only in the same scope.

I therefore remove this check.